### PR TITLE
Add Ansible job template selection

### PR DESCRIPTION
**Pull request by chenz4027**
https://github.com/open-cluster-management/backlog/issues/14699

### CODE REVIEW

File: frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.tsx [hunk 1]
```diff
@@ -475,24 +512,24 @@ export function AnsibleAutomationsForm(props: {
...
     ansibleJob?: ClusterCuratorAnsibleJob
     setAnsibleJob: (ansibleJob?: ClusterCuratorAnsibleJob, old?: ClusterCuratorAnsibleJob) => void
-    ansibleJobList?: ClusterCuratorAnsibleJob[]
 }) {
     const { t } = useTranslation(['common', 'cluster'])
     const [ansibleJob, setAnsibleJob] = useState<ClusterCuratorAnsibleJob | undefined>()
-    let ansibleJobList: string[]
-    if (props.ansibleJobList)
-        ansibleJobList = props.ansibleJobList.filter((job) => ansibleJob !== job).map((ansibleJob) => ansibleJob.name)
-    useEffect(() => setAnsibleJob(props.ansibleJob), [props.ansibleJob])
```
Comment (Randy424) [inline frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.tsx:495]: Sorry @chenz4027, I went a touch too far with this one.
This line `useEffect(() => setAnsibleJob(props.ansibleJob), [props.ansibleJob])`
actually needs to be here! It's what's currently braking the build.

Comment (chenz4027) [reply frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.tsx:495]: Ahh I see thanks for the catch!

File: frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.test.tsx [hunk 2]
```diff
@@ -74,13 +73,13 @@ const mockClusterCurator: ClusterCurator = {
     spec: {
         install: {
             towerAuthSecret: 'ansible-test-secret',
-            prehook: [{ name: 'test-job-pre-install', extra_vars: {} }],
```
Comment (jamestalton) [inline frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.test.tsx:77]: these names are not not unique between sections

File: frontend/package.json [hunk 3]
```diff
@@ -106,6 +106,7 @@
             "@typescript-eslint/ban-types": "off",
             "no-empty": "off",
             "no-constant-condition": "off",
+            "no-debugger": "off",
```
Comment (jamestalton) [inline frontend/package.json:109]: this needs to be removed

File: frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.test.tsx [hunk 4]
```diff
@@ -74,13 +73,13 @@ const mockClusterCurator: ClusterCurator = {
...
-            posthook: [{ name: 'test-job-post-install', extra_vars: {} }],
```
Comment (jamestalton) [inline frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.test.tsx:78]: these are still not unique between stages.  We has a bug on this at one point and the job names should be unique across stages.  i.e. spec.install.prehook[0].name should not equal spec.upgrade.prehook[0].name

**Review by jamestalton — APPROVED** [review]
lgtm

**Review by Randy424 — APPROVED** [review]

**Pull request merged**